core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2067,8 +2067,8 @@ class SparkContext(config: SparkConf) extends Logging {
   }
 
   /**
-   * Run a job on all partitions in an RDD and pass the results to a handler function, but take
-   * a function of type `Iterator[T] => U` instead of `(TaskContext, Iterator[T]) => U`.
+   * Run a job on all partitions in an RDD and pass the results to a handler function, this time the
```

Comment:
Sure, sounds more natural to me. Instead of "should", say "... each partition additionally takes ..." because it's not optional.